File: Meetups/WWC/arrays/delete_zeroes.py
"""
Given a list of integers, remove all the zeros
"""

# list comprehension
def remove_zeroes(arr, val):
    arr[:] = [i for i in arr if i != val]
    return arr

print remove_zeroes([1, 3, 0, 0, 5, 0, 7, 0, 0, 0, 9], 0)

# remove(). not optimal solution?
def remove_val(arr, val):
    while val in arr:
        arr.remove(val)
    return arr

print remove_val([1, 0, 0, 2, 0, 5], 0)

# lambda and filter
x = [0, 0, 5, 9, 0, 8, 0, 0]
print filter(lambda a: a != 0, x)

File: WWC/arrays/delete_zeroes.js
// Given a list, remove all zeros

// use splice
function delete_zeroes(list, val) {
    var index = list.length;
    console.log("length", index);
    while (index--) {
        console.log(index);
        if (list[index] == val) {
            list.splice(index,1);
        }
    }
    return list;
}

console.log(delete_zeroes([0, 1, 0, 0, 9, 8, 7, 0, 0], 0));
